benchmark/distributed/master.js

```diff
@@ -0,0 +1,163 @@
+  })
+  .help()
+  .alias('help', 'h')
+  .argv;
+
+const {
+  address,
+  report: reportInterval,
+  key,
+  cert,
+} = args;
+
+const { transport, listen: listenAddress } = parseAddress(address);
+
+let startTimeHR = null;
+let workersReportBuffer = [];
+const workersReports = [];
+const serverReports = [];
+
+const workers = [];
+let server = null;
+let master = null;
+
+let serverAddress = null;
+
+const application = {
+  master: {
+    registerServer: (connection, address, callback) => {
+      if (server) {
+        callback('Server is already registered');
+      }
+      server = connection;
+      serverAddress = address;
+      workers.forEach((worker) => {
+        worker.emitRemoteEvent('worker', 'connect', [serverAddress]);
+      });
+      connection.on('event', (iface, event, args) => {
+        if (iface !== 'master') {
+          return;
+        }
+        if (event === 'report') {
+          const count = +args[0];
+          serverReports.push(args);
+          console.log(`Server counnections: ${count}`);
+        }
+      });
+      callback(null);
+    },
+
+    registerWorker: (connection, callback) => {
+      workers.push(connection);
+      if (serverAddress) {
+        connection.emitRemoteEvent('master', 'connect', [serverAddress]);
+      }
+
+      connection.on('event', (iface, event, args) => {
+        if (iface !== 'master') {
+          return;
+        }
+        if (event === 'report') {
+          const requests = +args[0];
+          workersReportBuffer.push(requests);
+        }
+      });
+      callback(null);
+    },
+
+    deregisterWorker: (connection, callback) => {
+      const index = workers.findIndex(conn => conn === connection);
+      if (index === -1) {
+        callback('Not registered');
+        return;
+      }
+      workers.splice(index, 1);
+      callback(null);
+    },
+  },
+};
+
+const config = {
+  applications: [new jstp.Application('master', application)],
+  key,
+  cert,
+};
+
+// Prevent Node.js from throwing DEPTH_ZERO_SELF_SIGNED_CERT.
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+
+master = jstp[transport].createServer(config);
+
+master.listen(listenAddress, () => {
+  console.log(`Master server listening on ${address}`);
+  startTimeHR = process.hrtime();
+
+  let intervalTimerHR = process.hrtime();
+  const rpsReports = [];
+  setTimeout(() => {
+    setInterval(() => {
+      const timePassedHR = process.hrtime(intervalTimerHR);
+      const timePassed = timePassedHR[0] * 1e9 + timePassedHR[0];
+      const requests = workersReportBuffer.reduce(
+        (sum, current) => sum + current, 0
+      );
+      workersReportBuffer = [];
+      workersReports.push([requests, process.hrtime(startTimeHR)]);
+      const rps = requests * 1e9 / timePassed;
+      rpsReports.push(rps);
+      const mean = statistics.mean(rpsReports);
+      const stdev = statistics.stdev(rpsReports, mean);
+      console.log(`Current RPS: ${rps}, mean: ${mean}, stdev: ${stdev}`);
```

Comment:
IMHO, there sis no reason at all to show it here. It will always say 10±1 second.